File: lab-redes/python/tcp/cliente_tcp.py
import socket

HOST = "localhost"
PORTA = 5022

with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as cliente:
    cliente.connect((HOST, PORTA))
    print("[TCP] Conectado ao servidor. Digite 'sair' para encerrar.")
    arquivo = cliente.makefile("r")

    while True:
        mensagem = input("> ")
        cliente.sendall((mensagem + "\n").encode("utf-8"))
        print(arquivo.readline().strip())
        if mensagem.lower() == "sair":
            break

File: lab-redes/java/tcp/ClienteTCP.java
import java.io.*;
import java.net.*;

public class ClienteTCP {
    public static void main(String[] args) throws IOException {
        String host = "localhost";
        int porta = 5022;

        try (Socket socket = new Socket(host, porta);
             PrintWriter saida = new PrintWriter(socket.getOutputStream(), true);
             BufferedReader entrada = new BufferedReader(
                     new InputStreamReader(socket.getInputStream()));
             BufferedReader teclado = new BufferedReader(new InputStreamReader(System.in))) {

            System.out.println("[TCP] Conectado ao servidor. Digite 'sair' para encerrar.");
            String linha;
            while (true) {
                System.out.print("> ");
                linha = teclado.readLine();
                saida.println(linha);
                System.out.println(entrada.readLine());
                if (linha.equalsIgnoreCase("sair")) break;
            }
        }
    }
}